docs/resources/process_automation_trigger.md

```diff
@@ -0,0 +1,60 @@
+---
+page_title: "genesyscloud_process_automation_trigger Resource - terraform-provider-genesyscloud"
+subcategory: ""
+description: |-
+  Genesys Cloud Process Automation Trigger
+  NOTE: This component is currently in beta. If you wish to use this provider make sure your client has the correct permissions
+---
+# genesyscloud_process_automation_trigger (Resource)
+
+Genesys Cloud Process Automation Trigger
+
+**NOTE: This component is currently in beta. If you wish to use this provider make sure your client has the correct permissions**
+
+## API Usage
+The following Genesys Cloud APIs are used by this resource. Ensure your OAuth Client has been granted the necessary scopes and permissions to perform these operations:
+
+* [GET /api/v2/processAutomation/triggers](https://developer.genesys.cloud/platform/preview-apis#get-api-v2-processautomation-triggers)
+* [POST /api/v2/processAutomation/triggers](https://developer.genesys.cloud/platform/preview-apis#post-api-v2-processautomation-triggers)
+* [GET /api/v2/processAutomation/triggers/{triggerId}](https://developer.genesys.cloud/platform/preview-apis#get-api-v2-processautomation-triggers--triggerId-)
+* [PUT /api/v2/processAutomation/triggers/{triggerId}](https://developer.genesys.cloud/platform/preview-apis#put-api-v2-processautomation-triggers--triggerId-)
+* [DELETE /api/v2/processAutomation/triggers/{triggerId}](https://developer.genesys.cloud/platform/preview-apis#delete-api-v2-processautomation-triggers--triggerId-)
+
+## Example Usage
+
+```terraform
+resource "genesyscloud_process_automation_trigger" "test-trigger" {
+  name       = "Test Trigger"
+  topic_name = "v2.detail.events.conversation.{id}.customer.end"
+  enabled    = true
+  target = jsonencode({
+    id : "ae1e0cde-875d-4d13-a498-615e7a9fe956",
+    type : "Workflow"
+  })
+  match_criteria = jsonencode([
```

Comment:
Does match_criteria always have these 3 properties? Looks like it's a `MatchCriteria[]`. Therefore it should be of type Set containing a `schema.Resource` in the Elem value. You can see other nested objects like this in other resources.

Same goes for the `target` attribute.

Reply:
The matchCriteria can change, it will. have either a value field or a values field. Target should always have the same 2 fields

Reply:
Yeah they should both be Sets then containing nested objects. value and values in matchCriteria can be optional